amlip_py/amlip_py/node/FiwareNode.py

```diff
@@ -0,0 +1,509 @@
+
+import logging
+from py_utils.logging.log_utils import CustomLogger
+from py_utils.wait.BooleanWaitHandler import BooleanWaitHandler
+from py_utils.wait.WaitHandler import AwakeReason
+
+from amlip_py.node.AsyncEdgeNode import AsyncEdgeNode, InferenceListenerLambda
+from amlip_py.types.InferenceDataType import InferenceDataType
+
+from amlip_swig import FiwareNode as cpp_FiwareNode
+
+
+headers_POST = {
+    'Content-Type': 'application/json'
+}
+
+headers_GET = {
+    'Accept': 'application/json'
+}
+
+
+class FiwareNode(cpp_FiwareNode):
+    """
+    AML-IP Fiware Node.
+
+    """
+
+    def __init__(
+            self,
+            name: str,
+            server_ip: str,
+            server_port: int,
+            context_broker_ip: str = 'localhost',
+            context_broker_port: int = 1026,
+            entity_id: str = 'ID_0',
+            entity_data: str = 'data',
+            entity_solution: str = 'inference',
+            domain: int = None,
+            logger=CustomLogger(logger_name='FiwareNode', log_level=logging.WARNING)):
+        """
+        Create a new Fiware Node with a given name.
+
+        Parameters
+        ----------
+        name : str
+            Name of the node.
+        server_ip : str
+            IP address of the server.
+        server_port : int
+            Port of the server.
+            Defaults to 1028.
+        context_broker_ip : str
+            IP address of the context broker.
+            Defaults to 'localhost'.
+        context_broker_port : int
+            Port of the context broker.
+            Defaults to 1026.
+        entity_id : str
+            ID of the context broker entity.
+            Defaults to 'ID_0'.
+        entity_data : str
+            Name of the entity data attribute.
+            Defaults to 'data'.
+        entity_solution : str
+            Name of the entity solution attribute.
+            Defaults to 'inference'.
+        logger : logging.Logger
+            Logger object.
+            Defaults to a new CustomLogger with log level WARNING.
+
+        """
+        #####
+        # Parent class constructor
+        if domain is None:
+            super().__init__(name)
+        else:
+            super().__init__(name, domain)
+
+        self.server_ip = server_ip
+        self.server_port = server_port
+        self.context_broker_ip = context_broker_ip
+        self.context_broker_port = context_broker_port
+        self.id = entity_id
+        self.entity_data = entity_data
+        self.entity_solution = entity_solution
+        self.logger = logger
+        self.logger.info(f'Creating FiwareNode with name: {name}')
+
+        self.name = name
+
+        self.edge = AsyncEdgeNode('edge_node'+name,
+                                  listener=InferenceListenerLambda(self.inference_received))
+
+        self.app = Flask(__name__)
+
+        self.setup_routes()
+        self.init_subscriptions()
+
+        self.waiter_inference = BooleanWaitHandler(True, True)
```

Comment:
I think we should create the waiter in a closed state. If we create it open and then call `get_inference()`, it might incorrectly assume that an inference has already arrived, which is not the case.